golemcli/src/commands/status.rs

```diff
@@ -0,0 +1,603 @@
+            })
+            .map(
+                |(total_subtasks, finished_subtasks)| match total_subtasks > 0 {
+                    true => Some(format!("{}/{}", finished_subtasks, total_subtasks)),
+                    false => None,
+                },
+            )
+            .from_err()
+    }
+}
+
+impl FormattedObject for FormattedGeneralStatus {
+    fn to_json(&self) -> Result<Value, Error> {
+        Ok(serde_json::to_value(&self)?)
+    }
+
+    fn print(&self) -> Result<(), Error> {
+        let mut section_builder = SectionBuilder::new( "  ");
+
+        section_builder
+            .new_section("General")
+            .entry(
+                "Process State",
+                &self.running_status.process_state.to_string(),
+            )
+            .new_subsection("Components Status")
+            .entry(
+                "docker",
+                self.running_status
+                    .component_statuses
+                    .docker_status
+                    .as_ref()
+                    .unwrap_or(&String::from("unknown")),
+            )
+            .entry(
+                "hyperdrive",
+                self.running_status
+                    .component_statuses
+                    .hyperdrive
+                    .as_ref()
+                    .unwrap_or(&String::from("unknown")),
+            )
+            .entry(
+                "client",
+                self.running_status
+                    .component_statuses
+                    .client
+                    .as_ref()
+                    .unwrap_or(&String::from("unknown")),
+            );
+
+        if self.running_status.component_statuses.hypervisor.is_some() {
+            section_builder.entry(
+                "hypervisor",
+                self.running_status
+                    .component_statuses
+                    .hypervisor
+                    .as_ref()
+                    .unwrap(),
+            );
+        }
+        let connection_status = if self.net_status.is_connected {
+            String::from("ONLINE")
+        } else {
+            String::from("OFFLINE")
+        };
+
+        section_builder
+            .end_section()
+            .entry("Golem version", &self.running_status.golem_version)
+            .entry("Node name", &self.running_status.node_name)
+            .entry("Network", &self.running_status.network.to_string())
+            .new_subsection("Disk usage")
+            .entry(
+                "Received files",
+                &self.running_status.disk_usage.received_files,
+            )
+            .entry(
+                "Distributed files",
+                &self.running_status.disk_usage.distributed_files,
+            )
+            .end_section()
+            .end_section()
+            .new_section("Network")
+            .entry("Connection", &connection_status)
+            .entry(
+                "Port statuses",
+                &format!("{:?}", self.net_status.port_status),
```

Comment:
this should be more human readable, not json

Reply:
done